src/cdk-experimental/menu/menu-base.ts

```diff
@@ -28,8 +28,11 @@ import {PointerFocusTracker} from './pointer-focus-tracker';
 
 @Directive({
   host: {
+    '[tabindex]': '_isInline ? (_hasFocus ? -1 : 0) : null',
     '[attr.aria-orientation]': 'orientation',
     '(focus)': 'focusFirstItem()',
+    '(focusin)': 'menuStack.setHasFocus(true)',
```

Comment:
One thing that you may want to double-check is what happens if there are focusable elements inside the menu. From what I remember, `focusin` and `focusout` events bubble so it might trip up the menu if focus moves from one child to another.

Reply:
Is that a thing we need to consider? My understanding was that the only interactive elements in a menu should be the menu items and that menu items can't contain interactive elements.

Reply:
That's true, but the `ng-content` implies that anything can be projected. Anyway, I don't think that it's a blocker, I just wanted to flag it.

Reply:
Sure, but I'm just not sure if we need to go out of our way to make it work if they project something invalid. I feel like our team has kind of a bad habit of trying to support every possible thing a user might do no matter how crazy. I'd prefer to focus on making a good experience for the people who are using it correctly. 

Though it's definitely a good point to remember when we're writing the docs, we do actually have to tell people which things are correct usage and which are not